src/database/InMemoryDatabase.ts

```diff
@@ -77,8 +77,24 @@ export class InMemoryDatabase implements Database {
     return Promise.resolve();
   }
 
-  deleteOrder(hash: string) {
-    delete this.database[hash];
+  deleteOrder(nonce: String, signerWallet: string): Promise<void> {
+    const orderToDelete = Object.values(this.database).filter((indexedOrder: IndexedOrder) => {
+      return indexedOrder.order.nonce === nonce && indexedOrder.order.signerWallet === signerWallet
+    });
+    console.log("orderToDelete", orderToDelete)
+    if (orderToDelete.length > 0) {
+      delete this.database[orderToDelete[0].hash as string];
+    }
+    return Promise.resolve();
+  }
```

Comment:
```suggestion
  deleteOrder(nonce: string, signerWallet: string): Promise<void> {
    const orderToDelete = Object.values(this.database).find((indexedOrder: IndexedOrder) => {
      return indexedOrder.order.nonce === nonce && indexedOrder.order.signerWallet === signerWallet
    });
    console.log("orderToDelete", orderToDelete);
    if (orderToDelete && orderToDelete.hash) {
      delete this.database[orderToDelete.hash];
    }
    return Promise.resolve();
  }
```

Also no Promise.reject if order is not found? Or do you intentionally want it to keep silent?

Reply:
Thank you, 
No need to throw if the order was not stored in the database
Maybe later, we could try to synchronise with others if a node detects that is missing some orders